.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseMergedParser.cs

```diff
@@ -98,6 +98,10 @@ public DateTimeParseResult Parse(ExtractResult er, DateObject refTime)
             {
                 pr = this.Config.DatePeriodParser.Parse(er, referenceTime);
             }
+            else if (er.Type.Equals(Constants.SYS_DATETIME_TIMEZONE))

```

Comment:
I think timezone should be the last condition, to avoid interfering with other pieces.
Any special reason it comes before period?

Reply:
Not sure about this part. I can move it to the last condition.

Reply:
Please do, if it doesn't cause any issue.